src/plugins/vis_types/timeseries/server/lib/vis_data/get_series_data.ts

```diff
@@ -117,7 +117,10 @@ export async function getSeriesData(
       [panel.id]: {
         annotations,
         id: panel.id,
-        series: series.reduce((acc, s) => acc.concat(s), []),
+        series: series.reduce((acc, s) => {
```

Comment:
I did profile this already and `concat` is way faster than `push` + spread.
In terms of GCC pressure both approach will create a new Array and clean up but the `concat` approach avoids to iterate on the second one.

Anyway I've created a benchmark test case to verify my previous findings and... the thing is "nuanced": https://jsbenchmark.com/#eyJjYXNlcyI6W3siaWQiOiI3d0loS0FkVWpWTGc4YVNCRUd3R1giLCJjb2RlIjoiY29uc3QgYyA9IERBVEEuc2VyaWVzLnJlZHVjZSggKG1lbW8sIGFycikgPT4ge1xuXHRyZXR1cm4gWy4uLm1lbW8sIC4uLmFycl07XG59LCBEQVRBLmEpOyIsIm5hbWUiOiJSZWR1Y2Ugd2l0aCBzcHJlYWQifSx7ImlkIjoiQW10S1F0TVNleDB3bDFnVXJLSmRFIiwiY29kZSI6ImNvbnN0IGMgPSBEQVRBLnNlcmllcy5yZWR1Y2UoIChtZW1vLCBhcnIpID0+IG1lbW8uY29uY2F0KGFyciksIERBVEEuYSk7IiwibmFtZSI6IlJlZHVjZSB3aXRoIGNvbmNhdCJ9LHsiaWQiOiI4cENpY1pTdTYwaTYxLURRTXdUV1kiLCJjb2RlIjoiY29uc3QgYyA9IERBVEEuc2VyaWVzLnJlZHVjZSggKG1lbW8sIGFycikgPT4ge1xuXHRtZW1vLnB1c2goLi4uYXJyKTtcblx0cmV0dXJuIG1lbW87XG59LCBEQVRBLmEpOyIsIm5hbWUiOiJSZWR1Y2Ugd2l0aCBwdXNoICsgc3ByZWFkIn0seyJpZCI6IlVhZFk1U0NDZl9RR05KZTEwSWpDTSIsImNvZGUiOiJjb25zdCBjID0gREFUQS5hLmNvbmNhdChEQVRBLnNlcmllcy5mbGF0KCkpIiwiZGVwZW5kZW5jaWVzIjpbXSwibmFtZSI6IkNvbmNhdCArIGZsYXQifV0sImNvbmZpZyI6eyJuYW1lIjoiU2ltcGxlIGV4YW1wbGUgdGVzdCIsInBhcmFsbGVsIjp0cnVlLCJnbG9iYWxUZXN0Q29uZmlnIjp7ImRlcGVuZGVuY2llcyI6W119LCJkYXRhQ29kZSI6InJldHVybiB7XG4gIGE6IEFycmF5LmZyb20oe2xlbmd0aDogMTAwMH0pLm1hcCgoXywgaSk9PmAke2l9YCksXG4gIHNlcmllczogQXJyYXkuZnJvbSh7bGVuZ3RoOiAxMH0pLmZpbGwoQXJyYXkuZnJvbSh7bGVuZ3RoOiAyMDAwfSkubWFwKChfLCBpKT0+YCR7aX1gKSlcbn07In19

Surprisingly `Chrome` fails to execute on the `push + spread` on my machine. But `concat` in Chrome is super fast.
Safari seems to be faster with `push + spread`, with `concat` second.
Firefox is way faster with `concat`.

They are still micro-benchmarks, so probably it is not worth push more here, but my point is that `concat` is "fast enough" already and there's no need to make the code more complex in this case.

Reply:
Have you tried on a real server-side environment? The results from my personal benchmarks are surprisingly different between chrome/firefox and nodejs.

Now if you think some changes are unnecessary, I'm fine with it. would that be the only file you want me to revert?

Reply:
Can you report some results from the nodejs benchmarks?

As I said, if the perf toll is not dramatically higher with `concat` I would rather keep this version that the other. Otherwise I think it is ok to change it.